### ISSUE TITLE
[Bug]: provider │ "registry.terraform.io/hashicorp/aws" produced an invalid new value for │ .container_definitions

### ISSUE THREAD

**Issue by giuliamorg92** [hashicorp/terraform-provider-aws#38564]
### Terraform Core Version

1.5.0,1.9.0

### AWS Provider Version

5.59.0

### Affected Resource(s)

aws_ecs_task_definition

### Expected Behavior

Task definition creation

### Actual Behavior

Provider produced inconsistent final plan

### Relevant Error/Panic Output Snippet

```shell
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_ecs_task_definition.ecs_task_definition to
│ include new values learned so far during apply, provider
│ "registry.terraform.io/hashicorp/aws" produced an invalid new value for
│ .container_definitions: was cty.StringVal(""), but now
│ cty.StringVal("[{\"Command\":null,\"Cpu\":256,\"CredentialSpecs\":null,\"DependsOn\":null,\"DisableNetworking\":null,\"DnsSearchDomains\":null,\"DnsServers\":null,\"DockerLabels\":null,\"DockerSecurityOptions\":null,\"EntryPoint\":null,\"Environment\":[{\"Name\":\"APPLICATION_NAME\",\"Value\":\"[redacted]\"},{\"Name\":\"ENVIRONMENT_NAME\",\"Value\":\"dev\"}],\"EnvironmentFiles\":null,\"Essential\":true,\"ExtraHosts\":null,\"FirelensConfiguration\":null,\"HealthCheck\":null,\"Hostname\":null,\"Image\":\[redacted],\"Interactive\":null,\"Links\":null,\"LinuxParameters\":null,\"LogConfiguration\":{\"LogDriver\":\"awslogs\",\"Options\":{\"awslogs-group\":\"[redacted]\",\"awslogs-region\":\"eu-west-1\",\"awslogs-stream-prefix\":\"ecs\"},\"SecretOptions\":null},\"Memory\":512,\"MemoryReservation\":null,\"MountPoints\":null,\"Name\":\[redacted]\",\"PortMappings\":[{\"AppProtocol\":\"\",\"ContainerPort\":3000,\"ContainerPortRange\":null,\"HostPort\":3000,\"Name\":null,\"Protocol\":\"\"}],\"Privileged\":null,\"PseudoTerminal\":null,\"ReadonlyRootFilesystem\":null,\"RepositoryCredentials\":null,\"ResourceRequirements\":null,\"Secrets\":[{\"Name\":\"envs\",\"ValueFrom\":\"arn:aws:secretsmanager:eu-west-1:[redacted]:secret:secrets/dev/[redacted]\"}],\"StartTimeout\":null,\"StopTimeout\":null,\"SystemControls\":null,\"Ulimits\":null,\"User\":null,\"VolumesFrom\":null,\"WorkingDirectory\":null}]").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
```

### Terraform Configuration Files

```
#Create task definitions for app services
resource "aws_ecs_task_definition" "ecs_task_definition" {
  family                   = "${var.env}-${var.project}-${var.app_name}"
  task_role_arn            = aws_iam_role.ecs_role.arn
  execution_role_arn       = aws_iam_role.ecs_task_execution_role.arn
  requires_compatibilities = ["FARGATE"]
  network_mode             = "awsvpc"
  cpu                      = var.cpu
  memory                   = var.memory
  container_definitions = jsonencode([
    {
      name        = "${var.env}-${var.project}-${var.app_name}-task"
      image       = var.ecr_repo_uri != "" ? "${var.ecr_repo_uri}:${var.ecr_image_tag}" : (var.is_create_repo ? "${aws_ecr_repository.repo[0].repository_url}:latest" : "")      
      cpu         = "${var.cpu}"
      memory      = "${var.memory}"
      environment = var.environment_variables
      essential   = true
      secrets     = var.secrets_enable == true ? concat(
        [
          {
            name      = "envs",
            valueFrom = aws_secretsmanager_secret.secrets[0].arn
          }
        ],
      ) : null
      portMappings = [
        {
          containerPort = var.https_tg_port
          hostPort      = var.https_tg_port
        }
      ]
      logConfiguration = {
        logDriver = "awslogs"
        options = {
          "awslogs-group"         = aws_cloudwatch_log_group.ecs_cw_log_group.name
          "awslogs-region"        = "eu-west-1"
          "awslogs-stream-prefix" = "ecs"
        }
      }
    }
  ])
    dynamic "runtime_platform" {
    for_each = var.runtime_platform
    content {
      operating_system_family = runtime_platform.value.operating_system_family
      cpu_architecture        = runtime_platform.value.cpu_architecture
    }
  }
}

```

### Steps to Reproduce

Run `terraform apply` with provider version <= 5.58 creates the task definition correctly, upgrading to 5.59 fails due to inconsistent plan.

### Would you like to implement a fix?

None

**Comment by ewbankkit** [comment]
@giuliamorg92 Thanks for raising this issue.
It has already been noticed in #38461. I'm going to close this one as a duplicate so that we can concentrate discussion in the linked issue.
Please add any additional comments there.

**Comment by ewbankkit** [comment]
The fix (https://github.com/hashicorp/terraform-provider-aws/pull/38471) is available in Terraform AWS Provider **v5.60.0**.
